accessible/toolbox_treeview.component.js

```diff
@@ -0,0 +1,189 @@
+        </li>
+        <li #sendToSelected id='{{treeService.createId(sendToSelected)}}' role='treeitem' aria-selected=false [attr.aria-level]='level+2'>
+          <button #sendToSelectedButton id='{{treeService.createId(sendToSelectedButton)}}' (click)='copyToMarked(block)' disabled='{{notCompatibleWithMarkedBlock(block)}}' [attr.aria-disabled]='notCompatibleWithMarkedBlock(block)'>copy to marked spot</button>
+          {{setLabelledBy(sendToSelected, concatStringWithSpaces(sendToSelectedButton.id, 'blockly-button', notCompatibleWithMarkedBlock(block)))}}
+        </li>
+      </ol>
+      {{addClass(listItem, 'blocklyHasChildren')}}
+    </li>
+    <div *ngFor='#inputBlock of block.inputList; #i=index'>
+      <field-view [attr.aria-level]='level+1' *ngFor='#field of getInfo(inputBlock); #j=index' [field]='field' [level]='level+1'></field-view>
+      <toolbox-tree-view *ngIf='inputBlock.connection && inputBlock.connection.targetBlock()' [block]='inputBlock.connection.targetBlock()' [displayBlockMenu]='false' [level]='level+1'></toolbox-tree-view>
+      <li aria-selected=false #listItem1 role='treeitem' [attr.aria-level]='level+1' id='{{treeService.createId(listItem1)}}' *ngIf='inputBlock.connection && !inputBlock.connection.targetBlock()'>
+        <label #label id='{{treeService.createId(label)}}'>{{inputType(inputBlock.connection)}} {{valueOrStatement(inputBlock)}} needed:</label>
+        {{setLabelledBy(listItem1, concatStringWithSpaces('blockly-argument-text', label.id))}}
+      </li>
+    </div>
+  </ol>
+</li>
+<toolbox-tree-view *ngIf= 'block.nextConnection && block.nextConnection.targetBlock()' [level]='level' [block]='block.nextConnection.targetBlock()' [displayBlockMenu]='false'></toolbox-tree-view>
+    `,
+    directives: [ng.core.forwardRef(
+        function() { return blocklyApp.ToolboxTreeView; }), blocklyApp.FieldView],
+    inputs: ['block', 'displayBlockMenu', 'level', 'index', 'tree'],
+  })
+  .Class({
+    constructor: [blocklyApp.ClipboardService, blocklyApp.TreeService, function(_service, _service2) {
+      this.infoBlocks = {};
+      this.nextBlock = {};
+      this.sharedClipboardService = _service;
+      this.treeService = _service2;
+    }],
+    setLabelledBy: function(item,string){
+      if (!item.getAttribute('aria-labelledby')) {
+        item.setAttribute('aria-labelledby', string);
+      }
+    },
+    concatStringWithSpaces: function(){
+      var string = arguments[0];
+      for (i = 1; i < arguments.length; i++) {
+        var arg = arguments[i];
+        //arg can be undefined if it is coming from a function that checks whether the button should be disabled or not.
+        if (arg){
+          string = string + ' ' + arguments[i];
+        }
+      }
+      return string;
+    },
+    createCategoryDependantId: function(index, parentList){
+      //if this is the first block in a category-less toolbox, the id should be toolbox-tree-node0
+      if (index != undefined && index == 0) {
```

Comment:
Danger.  JavaScript does strange things with undefined, 0, and other falsy values.  I think this code is ok, but please change to !== and === for safety.